### PR TITLE
[One Page/Playable] Use l4zjs instead of node-lz4 for engine compression

### PR DESCRIPTION
**Pull request by yaustar**
Fixes: https://github.com/playcanvas/playcanvas-rest-api-tools/issues/46

lz4js (https://github.com/Benzinga/lz4js) is pure JS solution for compressing and uncompressing lz4 data. Despite being 8 years old, still works with with Node v20 and has no dependencies to native libs/code.

I couldn't use lz4-napi because it required the compiled libraries for both compressing and decompressing so it couldn't be packaged in the playable ad.

I also couldn't work out a way to use the existing node-lz4 without the native code bindings in Node.js so ended up finding lz4js which is pure JS.

Decompression time of lz4js is comparable to node-js, tested on a M1 Mac at 6x slowdown:

* lz4js averages at ~350ms
* node-lz4 averages at ~380ms

lz4js is also ~150KB smaller minified for the packaged build.

I've tested an ad with FB Ad CMS to ensure the new compression library works in their tester.

This PR also adds steps on how to build the browserify version of lz4js to use with the playable ad.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "base64-js": "^1.5.1",
     "btoa": "^1.2.1",
     "dotenv": "^8.2.0",
-    "lz4": "^0.6.5",
+    "lz4js": "^0.2.0",
```
Comment (yaustar) [inline package.json:21]: lz4js is ISC licensed based on its package.json

File: shared.js [hunk 2]
```diff
@@ -177,7 +177,7 @@ function unzipProject(zipLocation, unzipFolderName) {
         try {
             var tempFolder = path.resolve(path.dirname(zipLocation), unzipFolderName);
             if (fs.existsSync(tempFolder)) {
-                fs.rmdirSync(tempFolder, {recursive:true});
+                fs.rmSync(tempFolder, {recursive:true});
```
Comment (yaustar) [inline shared.js:180]: Updated deprecated function calls for Node v20+

File: shared.js [hunk 3]
```diff
@@ -198,7 +198,7 @@ function zipProject(rootFolder, targetLocation) {
             fs.mkdirSync(path.dirname(output));
         }
         zip.writeZip(output);
-        fs.rmdirSync(rootFolder, {recursive:true});
+        fs.rmSync(rootFolder, {recursive:true});
```
Comment (yaustar) [inline shared.js:201]: Updated deprecated function calls for Node v20+

File: library-files/lz4-browserify/package.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "lz4-browsify",
+  "version": "1.0.0",
+  "description": "Packages needed to build a browsify version of lz4js with Buffer",
```
Comment (yaustar) [inline library-files/lz4-browserify/package.json:4]:
```suggestion
  "description": "Packages needed to build a browserify version of lz4js with Buffer support",
```

File: library-files/lz4-browserify/reademe.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# How to build the Browserify version of lz4js
+
+Requires Node.js v20.
+
+1. Install NPM packages via `npm i`
+2. Install Browserify globablly `npm install -g browserify`
+3. Install Terser globally `npm i -g terser`
+4. Build the minified `browserify main.js | terser --compress > ../lz4.js`
```
Comment (yaustar) [inline library-files/lz4-browserify/reademe.md:8]:
```suggestion
4. Build the minified bundle `browserify main.js | terser --compress > ../lz4.js`
```

File: one-page.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ const base64js = require('base64-js');
 const { minify } = require('terser');
 const btoa = require('btoa');
 const replaceString = require('replace-string');
-const lz4 = require('lz4');
+const lz4 = require("lz4js");
```
Comment (yaustar) [inline one-page.js:7]:
```suggestion
const lz4 = require('lz4js');
```

**Review by willeastcott — COMMENTED** [review]
LGTM. So good to have this fixed. 😄

**Review by willeastcott — APPROVED** [review]

**Pull request merged**